services/QuillLMS/spec/controllers/learn_worlds_integration/webhooks_controller_spec.rb

```diff
@@ -0,0 +1,70 @@
+# frozen_string_literal: true
+
+require 'rails_helper'
+
+RSpec.describe LearnWorldsIntegration::WebhooksController, type: :controller do
+  include_context 'LearnWorlds Enrolled Free Course Event'
+  include_context 'LearnWorlds Course Completed Event'
+  include_context 'LearnWorlds Earned Certificate Event'
+
+  subject { post :create, params: params, as: :json }
+
+  let(:webhook_signature) { SecureRandom.hex(12) }
+
+  before { stub_const("Auth::LearnWorlds::WEBHOOK_SIGNATURE", webhook_signature) }
+
+  context '#create' do
+    before { allow(request).to receive(:env).and_return({described_class::SIGNATURE_HEADER_KEY => signature_header}) }
+
+    context 'invalid signature' do
+      let(:params) { {} }
+      let(:signature_header) { 'v2=invalid-signature' }
+      let(:error) { LearnWorldsIntegration::WebhooksController::InvalidSignatureError }
+
+      it 'handles the signature error and reports to new relic' do
+        expect(ErrorNotifier).to receive(:report).with(error)
+        subject
+        expect(response.status).to eq 400
```

Comment:
This is kind of nit-picky, but is an invalid signature a failure to authenticate?  Should this be a 403?

Reply:
I reused the stripe [structure](https://github.com/empirical-org/Empirical-Core/blob/develop/services/QuillLMS/app/controllers/stripe_integration/webhooks_controller.rb#L16) which was the code recommended in the stripe [docs](https://stripe.com/docs/payments/handling-payment-events).  Since the rescue clause is kind of bucket for any standard error I think it's more about sending a message back to the webhook that something failed.  While I agree with you point about it technically being the wrong code, I think for simplicity I'd rather just have the one clause for now.